From e115a1992ffed453cebf0bcb1ea3ea226fb4e0a8 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Fri, 6 May 2016 18:47:57 -0400
Subject: [PATCH 0471/1722] drm/amd/dal: Resolve error in
 drm_calc_vbltimestamp_from_scanoutpos.

crtc->hw_mode.clock whic is being checked for vblank counter and
TS calculations  is is not set in DRM unless calling drm_crtc_helper_set_mode
which we don't call in atomic mode_set.
This is a hack, a proper fix should be in DRM.

Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index b6e8579..2f1c069 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -2191,6 +2191,7 @@ int amdgpu_dm_atomic_commit(
 			acrtc->target = new_target;
 			acrtc->enabled = true;
 			acrtc->hw_mode = crtc->state->mode;
+			crtc->hwmode = crtc->state->mode;
 
 			break;
 		}
-- 
2.7.4

